### DISCUSSION TITLE
Follow up question about the boundary flux

### DISCUSSION THREAD

**Discussion by emmelines** [idaholab/moose#18711] [Q&A Modules: Phase field] [answered]
Hi 

I am simulating the microstructure evolution of the polycrystal two-phase separate system. I combined the Allen-Cahn and Cahn-Hillard equation into my system. 
The boundary condition I set is NeumannBC. But from the video, you could see the upper grains disappeared, and the lower grain coarsen. So that is the reason I bring up the question for boundary condition did not work.
I agreed with the numerical error @jiangwen84 mention. (Thank you. I appreciated it.)
However, the problem I faced is a flux systematic non-zero, and it alternated the conclusion of the simulation.
Is there anything I should check?

[-3.avi.zip](https://github.com/idaholab/moose/files/7062866/-3.avi.zip)

Thanks,

Emmeline

**Comment by hugary1995** [comment, marked as answer]
> But from the video, you could see the upper grains disappeared, and the lower grain coarsen. So that is the reason I bring up the question for boundary condition did not work.

No, this isn't the correct logic to me. If you apply zero flux, then all you get is global conservation of concentration. This is indeed observed in your video.

In your video, the movement of the interface is likely driven by something in your free energy, instead of the Neumann BC.
